PowerEditor/src/Parameters.cpp

```diff
@@ -4020,6 +4020,31 @@ void NppParameters::feedGUIParameters(TiXmlNode *node)
 			}
 		}
 
+		else if (!lstrcmp(nm, TEXT("SmartHighLightMode")))
+		{
+			TiXmlNode *n = childNode->FirstChild();
+
+			if (n)
+			{
+				const TCHAR* val = n->Value();
+				if (val)
+				{
+					if (!lstrcmp(val, TEXT("1")))

```

Comment:
Is this if/else block necessary (and subsequent checks when writing the parameter)? Can the pure enum value be stored instead?

If you take a look at the "_autoCAction" attribute in this file it uses its enum value.